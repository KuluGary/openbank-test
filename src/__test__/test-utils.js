import React from "react"
import { render as rtlRender } from "@testing-library/react"
import { Provider } from "react-redux"

import reduxStore from "store";

/**
 * Wraps the component in a mock of the redux store provider
 * @param {Component} ui - the UI component passed on render
 * @param {Object} renderOptions - the render options passed 
 */
const render = (ui, { initialState, store = reduxStore, ...renderOptions } = {}) => {
    const Wrapper = ({ children }) => {
        return <Provider store={store}>{children}</Provider>
    }

    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react"; // export full testing-library components
export { render }; // export provider wrapped render
