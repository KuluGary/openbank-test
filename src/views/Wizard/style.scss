@import "../../styles/base";

.wizard {
    max-width: 75vw;
    margin: 2rem auto;
    z-index: 100;

    @media (max-width: 768px) {
        max-width: 100vw;
        margin: 0;
    }
}

.button-container {
    padding: 2rem 3rem;
    display: flex;
    justify-content: space-between;
}