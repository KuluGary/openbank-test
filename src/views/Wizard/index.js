import React from "react";
import { Paper, Divider } from "@material-ui/core";
import Stepper from "../Stepper";
import Step1 from "../Steps/Step1";
import Step2 from "../Steps/Step2";
import Step3 from "../Steps/Step3";
import Buttons from "../Buttons";
import { useHistoryState } from "hooks/useHistoryState";
import "./style.scss";

/**
 * Component that controls all three main sub-components
 * of the wizard
 */

export default function Wizard() {
  // To add a new step, import it and include it in the
  // steps array in the correct index
  const steps = [Step1, Step2, Step3];
  const [validator, setValidator] = React.useState(false);
  const [activeStep, setActiveStep] = useHistoryState("step", 0);
  const SelectedComponent = steps[activeStep];

  const handleNext = () => {
    setActiveStep(activeStep + 1);
  };

  const handleBack = () => {
    setActiveStep(activeStep - 1);
  };

  return (
    <Paper className="wizard" data-testid="wizard">
      <Stepper steps={steps} activeStep={activeStep} />

      <SelectedComponent validator={validator} setValidator={setValidator} />

      <Divider />
      <div className="button-container">
        <Buttons
          activeStep={activeStep}
          handleBack={handleBack}
          handleNext={handleNext}
          validator={validator}
        />
      </div>
    </Paper>
  );
}
