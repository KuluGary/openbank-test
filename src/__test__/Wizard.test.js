import React from "react";
import Wizard from "views/Wizard";

import { render, cleanup } from "./test-utils";
import "@testing-library/jest-dom/extend-expect";
import { fireEvent } from "@testing-library/react";

describe("Wizard", () => {
  afterEach(cleanup);

  it("matches snapshot", () => {
    const div = document.createElement("div");
    const component = render(<Wizard />, div);
    expect(component.container).toMatchSnapshot();
  });

  it("renders component", () => {
    const { getByTestId } = render(<Wizard />);

    expect(getByTestId("wizard")).toBeInTheDocument();
  });

  it("navigates from step1 to step2", () => {
    const { getByTestId } = render(<Wizard />);

    expect(getByTestId("step1")).toBeInTheDocument();

    fireEvent.click(getByTestId("privacy-checkbox"));
    fireEvent.click(getByTestId("right-button"));

    expect(getByTestId("step2")).toBeInTheDocument();
  });
});
