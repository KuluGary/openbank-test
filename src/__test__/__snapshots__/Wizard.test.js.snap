// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wizard matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root wizard MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="wizard"
  >
    <div
      class="stepper"
      data-testid="stepper"
    >
      <div
        class="MuiPaper-root MuiStepper-root stepper-container MuiStepper-horizontal MuiPaper-elevation0"
      >
        <div
          class="MuiStep-root MuiStep-horizontal"
          data-testid="step"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  1
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            />
          </span>
        </div>
        <div
          class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
        >
          <span
            class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
          />
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal"
          data-testid="step"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  2
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            />
          </span>
        </div>
        <div
          class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
        >
          <span
            class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
          />
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal"
          data-testid="step"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  3
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="step1"
      data-testid="step1"
    >
      <h1>
        wizard.header
      </h1>
      <div
        class="step1-wrapper"
      >
        <div
          class="step1-wrapper-image-container"
        >
          <img
            alt="Protect your password"
            src="group.svg"
          />
          <p>
            wizard.step1.protect_password
          </p>
        </div>
        <div
          class="step1-wrapper-image-container"
        >
          <img
            alt="Protect your password"
            src="group-3.svg"
          />
          <p>
            wizard.step1.master_key
          </p>
        </div>
      </div>
      <h4>
        wizard.step1.how_it_works_header
      </h4>
      <p>
        wizard.step1.how_it_works
      </p>
      <h4>
        wizard.step1.which_data_header
      </h4>
      <p>
        wizard.step1.which_data
      </p>
      <div
        class="step1-checkbox"
      >
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-4"
                data-indeterminate="false"
                data-testid="privacy-checkbox"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            wizard.step1.checkbox
          </span>
        </label>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="button-container"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        data-testid="left-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          wizard.buttons.cancel
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary Mui-disabled Mui-disabled"
        data-testid="right-button"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          wizard.buttons.next
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
